App/Redux/CameraRollRedux.ts

```diff
@@ -85,6 +109,95 @@ export function reducer (state: CameraRollState = initialState, action: CameraRo
       )
       return { ...state, queriedPhotos }
     }
+    case getType(actions.imagePickerSuccess):
+      const { image } = action.payload
+      const threadShares = state.pendingShares
+
+      if (threadShares[action.payload.threadId] === undefined) {
+        // add the image uir to the thread's pending shares
+        threadShares[action.payload.threadId] = [image]
+      } else {
+        if (state.pendingShares[action.payload.threadId].find((img) => img.origURL === image.origURL)) {
+          // if the image already exists as a pending invite to the thread... skip it
+          return state
+        } else {
+          // add the image to the thread's pending shares
+          threadShares[action.payload.threadId].push(image)
+        }
+      }
+      return {...state, pendingShares: threadShares}
+    case getType(actions.addComment):
+      if (state.pendingShares[action.payload.threadId] === undefined) {
+        return state
+      }
+      const existing = state.pendingShares[action.payload.threadId].find(img=>img.origURL === action.payload.image.origURL)
+      if (!existing || existing.caption !== undefined) {
+        return state
+      }
+      existing.caption = action.payload.caption
+
+      const thread = state.pendingShares[action.payload.threadId].filter((img) => img.origURL !== action.payload.image.origURL)
+      thread.push(existing)
+
+      return {
+        ...state,
+        pendingShares: {
+          ...state.pendingShares,
+          [action.payload.threadId]: thread
+        }
+      }
+    case getType(actions.localPinSuccess):
+      if (state.pendingShares[action.payload.threadId] === undefined) {
+        return state
+      }
+      const partial = state.pendingShares[action.payload.threadId].find(img=>img.origURL === action.payload.image.origURL)
+      if (!partial || partial.addResult !== undefined) {
+        return state
+      }
+      partial.addResult = action.payload.addResult
+
+      const ps = state.pendingShares[action.payload.threadId].filter((img) => img.origURL !== action.payload.image.origURL)
+      ps.push(partial)
+
+      return {
+        ...state,
+        pendingShares: {
+          ...state.pendingShares,
+          [action.payload.threadId]: ps
+        }}
+    case getType(actions.remotePinSuccess):
+    case getType(actions.imagePinError):
```

Comment:
Not a very exciting Error handler, just removes it like a success. I added a small error 'Alert' in the thread view so that any error will tell the user something and move along